docs/yul.rst

```diff
@@ -976,6 +976,64 @@ within one Yul subobject. If at least one ``memoryguard`` call is found in a sub
 the additional optimiser steps will be run on it.
 
 
+verbatim
+^^^^^^^^
+
+The set of ``verbatim...`` builtin functions lets you create bytecode for opcodes
+that are not known to the Yul compiler. It also allows you to create
+bytecode sequences that will not be modified by the optimizer.
+
+The functions are ``verbatim_<n>i_<m>o("<data>", ...)``, where
+ - ``n`` is a decimal that specifies the number of input stack slots / variables
+ - ``m`` is a decimal that specifies the number of output stack slots / variables
+ - ``data`` is a string literal that contains the sequence of bytes
+
+If you for example want to define a function that multiplies the input
+by two, without the optimizer touching the constant two, you can use
+
+.. code-block:: yul
+
+    let x := calldataload(0)
+    let double := verbatim_1i_1o(hex"600202, x)
+
+This code will result in a ``dupN`` opcode to retrieve ``x``
+directly followed by ``600202``. The code is assumed to
+consume the copied value of ``x`` and produce the result
```

Comment:
```suggestion
directly followed by ``600202``. The compiler assumes that the copied value,
i.e., `dup1` in this case, is consumed and produces the output
```

Reply:
Should we also mention the order of the return variables? That is, for `let y, z := verbatim_1i_2o(....)`, the variable `z` is supposed to be on top of the stack, followed by `y`.

Reply:
good idea!